src/frontend/context/ConfigContext.tsx

```diff
@@ -95,81 +95,87 @@ const defaultContext: ConfigContextType = [
 
 const ConfigContext = React.createContext<ConfigContextType>(defaultContext);
 
-type Props = {
-  children: React.ReactNode;
-};
-
-export const ConfigProvider = ({ children }: Props) => {
+export const ConfigProvider = ({ children }: React.PropsWithChildren<{}>) => {
+  const mountedRef = React.useRef(false);
   const [config, setConfig] = React.useState(defaultConfig);
   const [status, setStatus] = React.useState<Status>("idle");
```

Comment:
This would be a change from current behaviour, but I think it makes more sense for this initial state to be `loading`, since on first mount it is always going to go into `loading` state after the first render. This would avoid a "flash" of `idle` state for components consuming this component.